SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialStatics.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialStatics.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "GeneralProjectSettings.h"
+#include "SpatialConstants.h"
+#include "SpatialGDKSettings.h"
+#include "Utils/ActorGroupManager.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialStatics);
+
+bool USpatialStatics::SpatialNetworkingEnabled()
+{
+    return GetDefault<UGeneralProjectSettings>()->bSpatialNetworking;
+}
+
+bool USpatialStatics::SpatialOffloadingEnabled()
+{
+    return SpatialNetworkingEnabled() && GetDefault<USpatialGDKSettings>()->bEnableOffloading;
+}
+
+bool USpatialStatics::IsAuthoritativeWorkerType(const AActor* Actor)
+{
+    if (Actor == nullptr)
+    {
+        UE_LOG(LogSpatialStatics, Warning, TEXT("Actor was nullptr in USpatialStatics::IsAuthoritativeWorkerType"));
```

Comment:
Why warning? If this is a precondition of the function then it should be an error? If nullptr is a valid input then it shouldn't warn.

Reply:
@Vatyx suggested a warning